src/components/PresenceMap/index.d.ts

```diff
@@ -0,0 +1,26 @@
+import { ReactNode } from 'react';
+import { BaseProps } from '../types';
+
+export interface IPosition {
+    lat: number;
+    lng: number;
+}
+
+export interface MapObject {
+    position: IPosition;
+    heading?: number;
+    image?: string | object;
+    onClick?: (event: MouseEvent<HTMLElement>) => void;
+}
+
+export interface PresenceMapProps extends BaseProps {
+    apiKey: string;
+    objects?: MapObject[];
+    style?: object;
```

Comment:
since PresenceMapProps extends form BaseProps style is not needed to redefine here